File: app/src/main/java/com/wilinz/yuetingmusic/ui/local/LocalListViewModel.java
package com.wilinz.yuetingmusic.ui.local;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.wilinz.yuetingmusic.data.model.Song;
import com.wilinz.yuetingmusic.data.repository.SongRepository;
import com.wilinz.yuetingmusic.ui.playlist.PlaylistViewModel1;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;

public class LocalListViewModel extends PlaylistViewModel1 {

    public LocalListViewModel(@NonNull Application application) {
        super(application);
    }

    @Override
    public Observable<List<Song>> getSongsList() {
        return SongRepository.getInstance().getLocalMusic(getApplication())
                .doOnNext(songs -> {
                    songsLiveData.postValue(songs);
                    refreshingLiveData.postValue(false);
                });
    }

}

File: app/src/main/java/com/wilinz/yuetingmusic/ui/local/LocalListViewModel.kt
package com.wilinz.yuetingmusic.ui.local

import android.app.Application
import com.wilinz.yuetingmusic.data.model.Song
import com.wilinz.yuetingmusic.data.repository.SongRepository
import com.wilinz.yuetingmusic.ui.playlist.PlaylistViewModel1
import io.reactivex.rxjava3.core.Observable

class LocalListViewModel(application: Application) : PlaylistViewModel1(application) {
    override val songsList: Observable<List<Song>>
        get() {
            return SongRepository.instance!!.getLocalMusic(getApplication())
                .doOnNext { songs: List<Song> ->
                    songsLiveData.postValue(songs)
                    refreshingLiveData.postValue(false)
                }
        }

}